src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -610,6 +616,29 @@ object IndexLogEntry {
   val VERSION: String = "0.1"
 
   def schemaString(schema: StructType): String = schema.json
+
+  /*
+   * Use this method to create a new IndexLogEntry, which automatically includes
+   * all common default hyperspace project properties.
+   * TODO: force dev to use this method as this takes into account all
+   *  project properties that needed to be added by default. Currently, dev can also
+   *  create IndexLogEntry from case class.
+   *  https://github.com/microsoft/hyperspace/issues/370
+   *  Also add require for hyperspace project version when we introduce breaking change.
+   */
+  def create(name: String,
+            derivedDataset: CoveringIndex,
+            content: Content,
+            source: Source,
+            properties: Map[String, String] = Map()): IndexLogEntry = {
```

Comment:
Let's not create a default value for this, similar to `IndexLogEntry` constructor, at least for this PR.

Reply:
fixed